src/Magento/FunctionalTestingFramework/StaticCheck/DeprecatedEntityUsageCheck.php

```diff
@@ -0,0 +1,605 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\StaticCheck;
+
+use Magento\FunctionalTestingFramework\Config\MftfApplicationConfig;
+use Magento\FunctionalTestingFramework\Exceptions\XmlException;
+use Magento\FunctionalTestingFramework\Page\Objects\SectionObject;
+use Magento\FunctionalTestingFramework\Test\Objects\ActionObject;
+use Magento\FunctionalTestingFramework\Page\Objects\ElementObject;
+use Magento\FunctionalTestingFramework\Test\Objects\ActionGroupObject;
+use Magento\FunctionalTestingFramework\Page\Objects\PageObject;
+use Magento\FunctionalTestingFramework\Test\Objects\TestObject;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Finder\Finder;
+use Exception;
+use Magento\FunctionalTestingFramework\Util\Script\ScriptUtil;
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\OperationDefinitionObjectHandler;
+use Magento\FunctionalTestingFramework\DataGenerator\Objects\OperationDefinitionObject;
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\DataObjectHandler;
+use Magento\FunctionalTestingFramework\DataGenerator\Objects\EntityDataObject;
+use Symfony\Component\Finder\SplFileInfo;
+use DOMNodeList;
+use DOMElement;
+
+/**
+ * Class DeprecatedEntityUsageCheck
+ * @package Magento\FunctionalTestingFramework\StaticCheck
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class DeprecatedEntityUsageCheck implements StaticCheckInterface
+{
+    const EXTENDS_REGEX_PATTERN = '/extends=["\']([^\'"]*)/';
+    const ACTIONGROUP_REGEX_PATTERN = '/ref=["\']([^\'"]*)/';
+
+    const ERROR_LOG_FILENAME = 'mftf-deprecated-entity-usage-checks';
+    const ERROR_LOG_MESSAGE = 'MFTF Deprecated Entity Usage Check';
+
+    /**
+     * Array containing all errors found after running the execute() function
+     *
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * String representing the output summary found after running the execute() function
+     *
+     * @var string
+     */
+    private $output;
+
+    /**
+     * ScriptUtil instance
+     *
+     * @var ScriptUtil
+     */
+    private $scriptUtil;
+
+    /**
+     * Data operations
+     *
+     * @var array
+     */
+    private $dataOperations = ['create', 'update', 'get', 'delete'];
+
+    /**
+     * Checks test dependencies, determined by references in tests versus the dependencies listed in the Magento module
+     *
+     * @param InputInterface $input
+     * @return string
+     * @throws Exception
+     */
+    public function execute(InputInterface $input)
+    {
+        $this->scriptUtil = new ScriptUtil();
+
+        $modulePaths = [];
+        $includeRootPath = true;
+        $path = $input->getOption('path');
+        if ($path) {
+            if (!realpath($path)) {
+                return "Invalid --path option: " . $path;
```

Comment:
when an invalid path is specified, it just prints the execution time with no error. Maybe throw an exception with the message.

Reply:
Fixed